services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -381,6 +337,148 @@ export class WorkflowService {
       },
     ];
   }
+  private async __copyFileAndCreate(
+    context: DefaultContextSchema,
+    entityId: TEntityId,
+  ): Promise<DefaultContextSchema> {
+    const documentsWithPersistedImages = await Promise.all(
+      context.documents.map(async document => ({
+        ...document,
+        pages: await this.__persistDocumentPagesFiles(document, entityId),
+      })),
+    );
+
+    return { ...context, documents: documentsWithPersistedImages };
+  }
+
+  private async __persistDocumentPagesFiles(
+    document: DefaultContextSchema['documents'][number],
+    entityId: string,
+  ) {
+    return await Promise.all(
+      document.pages.map(async documentPage => {
+        const id = `${document?.category}-${document?.type}-${document?.issuer?.country}`;
+
+        const documentContext =
+          `${document?.category}-${document?.type}-${document?.issuer?.country}`.toLowerCase();
+        const documentName = `${entityId}/${documentContext}_${crypto.randomUUID()}.${
+          documentPage.type
+        }`;
+
+        const { fromServiceProvider, fromRemoteFileConfig } =
+          this.__fetchFromServiceProviders(documentPage);
+        const { toServiceProvider, toRemoteFileConfig, remoteFileName } =
+          this.__fetchToServiceProviders(documentName);
+
+        const remoteFilePath = await this.fileService.copyFileFromSourceToSource(
+          fromServiceProvider,
+          fromRemoteFileConfig,
+          toServiceProvider,
+          toRemoteFileConfig,
+        );
+        const fileNameInBucket =
+          typeof remoteFilePath != 'string' ? remoteFilePath.fileNameInBucket : undefined;
+        const userId = entityId;
+        const ballerineFileId = await this.storageService.createFileLink({
```

Comment:
check if ballerineFileId exists, if it does - skip

Reply:
resolved